File: testFrameworks/android/UIAutomator/UIAutomatorCalculator/app/src/main/java/mart/io/uiautomatorcalculator/MainActivity.java
package mart.io.uiautomatorcalculator;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button) this.findViewById(R.id.button_1);

        final EditText firstOperand = (EditText) this.findViewById(R.id.input_field_1);
        final EditText secondOperand = (EditText) this.findViewById(R.id.input_field_2);

        final TextView resultView = (TextView) this.findViewById(R.id.result_field_2);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final float num1 = Float.parseFloat(firstOperand.getText().toString());
                final float num2 = Float.parseFloat(secondOperand.getText().toString());

                resultView.setText(String.valueOf(num1 * num2));
            }
        });
    }
}

File: android/udacity_kotlin_apps/AboutMe/app/src/main/java/com/example/aboutme/MainActivity.kt
package com.example.aboutme

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import com.example.aboutme.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private val myName: MyName = MyName("Aleks Mart")


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.myName = myName

        binding.editButton.setOnClickListener { addNickname(it) }
    }

    private fun addNickname(view: View) {

        binding.apply {
            myName?.nickname = nicknameTextId.text.toString()
            invalidateAll()

            nicknameTextId.visibility = GONE
            view.visibility = GONE
            nicknameTextResult.visibility = VISIBLE
        }

        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(view.windowToken, 0)
    }
}
